.github/workflows/release.yml

```diff
@@ -0,0 +1,50 @@
+---
+name: Publish release to GitHub and push package to PyPI
+on:
+  pull_request:
+    branches: [main]
+    tags:
+      - "v*.*.*"
+
+jobs:
+  build:
```

Comment:
on this release process, if I'm not wrong, we are just building and releasing, not actually testing.
Maybe we could use [workflow_run](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#workflow_run) to make sure that the release process happens after the test?